### DISCUSSION TITLE
Refit rest api does not seem to work in release build (NullReferenceException)

### DISCUSSION THREAD

**Discussion by JelleDamen** [reactiveui/refit#1498] [General]
Since I've updated my android project to .net6-android I'm not able to execute rest calls with refit in release anymore. It's pretty hard to debug, since it's only happening in the release build of the app. This is the stacktrace from appcenter:

### System.Net.Http.HttpClientHandler.SendAsync(HttpRequestMessage , CancellationToken )
System.NullReferenceException: Object reference not set to an instance of an object

```
System.Net.Http.HttpClientHandler.SendAsync(HttpRequestMessage , CancellationToken )
System.Net.Http.HttpMessageInvoker.SendAsync(HttpRequestMessage , CancellationToken )
System.Net.Http.HttpClient.<>n__0(HttpRequestMessage , CancellationToken )
System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage , HttpCompletionOption , CancellationTokenSource , Boolean , CancellationTokenSource , CancellationToken )
Refit.RequestBuilderImplementation.<>c__DisplayClass14_0`2.<<BuildCancellableTaskFuncForMethod>b__0>d[[PRO.Core.Models.AccessCodeResult, PRO.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[PRO.Core.Models.AccessCodeResult, PRO.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].MoveNext()
PRO.Core.Helpers.Execution.<retryWithApiResult>d__14`1[[PRO.Core.Models.AccessCodeResult, PRO.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].MoveNext()
```

I've think it's a linker issue, but I'm out of ideas to solve it. I also checked if HttpClient is the same in debug as in release (It's set to AndroidClientHandler).

Any ideas or suggestions I could try?

Thank you,

Jelle

**Comment by glennawatson** [comment]
This is a linker/AOT trimming symptom from the .NET 6 Android era (the request never reaches a configured handler, hence the NRE deep in `HttpClientHandler.SendAsync`). It is not reproducible against current Refit, and the toolchain has changed substantially since (modern `dotnet` Android, ILLink with better defaults).

If you still see this on a current setup, the usual fixes are:
- Explicitly set the handler (e.g. configure `HttpClient`/`HttpMessageHandler` via `HttpClientFactory`) rather than relying on the default.
- Ensure your models are preserved from trimming (Refit's source generator handles the interfaces; your DTOs may need `[DynamicallyAccessedMembers]`/trimming roots, or use the System.Text.Json source generator with a `JsonSerializerContext`).
- Test with the latest Refit and current .NET Android workload.

Closing as outdated. If it reproduces on a current toolchain, please open a new issue with a minimal repro and the exact SDK/Refit versions.
